STM/vertx/pom.xml

```diff
@@ -0,0 +1,199 @@
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+       Lesser General Public License for more details.
+
+       You should have received a copy of the GNU Lesser General Public
+       License along with this software; if not, write to the Free
+       Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+       02110-1301 USA, or see the FSF site: http://www.fsf.org. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.jboss.narayana.quickstart.stm</groupId>
+    <artifactId>stm-vertx-demo</artifactId>
+    <version>5.6.0.Final-SNAPSHOT</version>
+
+    <name>Narayana - STM and vertx demo </name>
+
+    <properties>
+        <trip.verticle>olddemo.Server</trip.verticle>
+        <test.verticle>reproducer.TripVerticle</test.verticle>
+        <demo2.verticle>olddemo.ServerVerticle</demo2.verticle>
+        <demo.verticle>olddemo.verticle.NonVolatileActivityVerticle</demo.verticle>
+        <!-- the main verticle class name -->
+        <main.verticle>${demo.verticle}</main.verticle>
+        <vertx.version>3.3.3</vertx.version>
+        <narayana.version>${project.version}</narayana.version>
+        <jboss-logging.version>3.1.4.GA</jboss-logging.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-core</artifactId>
+            <version>${vertx.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-unit</artifactId>
+            <version>3.3.3</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-web</artifactId>
+            <version>3.3.3</version>
+        </dependency>
+
+
+        <!-- Add hazelcast deps if you want it clusterable -->
+        <dependency>
+            <groupId>io.vertx</groupId>
+            <artifactId>vertx-hazelcast</artifactId>
+            <version>${vertx.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.narayana.stm</groupId>
+            <artifactId>stm</artifactId>
+            <version>${narayana.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging</artifactId>
+            <version>${jboss-logging.version}</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.7</version>
+        </dependency>
+    </dependencies>
+
+    <build>
```

Comment:
Needs a script to actually test each of the scenarios. I asked Ondra to do something similar for XTS SSL: https://github.com/jbosstm/quickstart/blob/master/XTS/ssl/pom.xml#L54